### PR TITLE
UI - keep path when changing reference in object view

### LINKED ISSUE #2441: [UI] Changing branch (Combobox) on the 'Objects' view resets the path
If the current path exists on the selected branch, you should keep it instead of reset into the root.

image

Taken from #2030

### PR DESCRIPTION
**Pull request by guy-har**
fixes #2441

### CODE REVIEW

File: webui/src/pages/repositories/repository/objects.jsx [hunk 1]
```diff
@@ -182,8 +182,8 @@ const ObjectsBrowser = ({ config, configError }) => {
                         withWorkspace={true}
                         selectRef={ref => router.push({
                             pathname: `/repositories/:repoId/objects`,
-                            params: {repoId: repo.id},
-                            query: {ref: ref.id}
+                            params: {repoId: repo.id, path:path},
+                            query: {ref: ref.id, path: path}
```
Comment (itaiad200) [inline webui/src/pages/repositories/repository/objects.jsx:186]: I get `undefined` as path if I'm switching between branches without any path:
image

**Review by itaiad200 — CHANGES_REQUESTED** [review]
See my comment around missing path.
Also - Do you think we should have the same behaviour for `Uncommitted Changes` and `Compare` tabs?

File: webui/src/pages/repositories/repository/objects.jsx (hunk 1 above)
Comment (guy-har) [reply webui/src/pages/repositories/repository/objects.jsx]: Oops, fixed it, thanks!

**Comment by guy-har** [comment]
@itaiad200 I am not sure it makes so much sense when it comes to `Uncommitted Changes` and `Compare`.
Maybe we should check with the user that requested this issue, WDYT?

**Comment by itaiad200** [comment]
> @itaiad200 I am not sure it makes so much sense when it comes to `Uncommitted Changes` and `Compare`. Maybe we should check with the user that requested this issue, WDYT?

`Compare` involves 2 branches and it's a different experience, I agree. `Uncommitted` is basically the same view but with only uncommitted objects presented. I think that the expectation would be that they would behave the same.

**Review by itaiad200 — APPROVED** [review]
LGTM

**Pull request merged**